src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1464,23 +1464,29 @@ private void DynamoViewModelRequestAboutWindow(DynamoViewModel model)
 
         private PublishPackageView _pubPkgView;
 
-        private void DynamoViewModelRequestRequestPackageManagerPublish(PublishPackageViewModel model)
+        private void DynamoViewModelRequestPackageManager(PublishPackageViewModel model)
         {
             var cmd = Analytics.TrackCommandEvent("PublishPackage");
-            if (_pubPkgView == null)
+
+            if (packageManagerWindow == null)
             {
-                _pubPkgView = new PublishPackageView(model)
+                packageManagerWindow = new PackageManagerView(this, _pkgVM)
                 {
                     Owner = this,
                     WindowStartupLocation = WindowStartupLocation.CenterOwner
                 };
-                _pubPkgView.Closed += (sender, args) => { _pubPkgView = null; cmd.Dispose(); };
-                _pubPkgView.Show();
 
-                if (_pubPkgView.IsLoaded && IsLoaded) _pubPkgView.Owner = this;
+                // setting the owner to the packageManagerWindow will centralize promts originating from the Package Manager
+                dynamoViewModel.Owner = packageManagerWindow;
+
+                packageManagerWindow.Closed += (sender, args) => { packageManagerWindow = null; cmd.Dispose(); };
```

Comment:
My last concern in this PR is that I recall that using anonymous event handler would cause memory leaks so maybe try using event handlers or lambda. You can refer to https://michaelscodingspot.com/5-techniques-to-avoid-memory-leaks-by-events-in-c-net-you-should-know/

Reply:
I made the change to explicitly subscribe and unsubscribe during the `WindowClosed` event, @QilongTang